Foundation/NSXMLParser.swift

```diff
@@ -254,10 +249,8 @@ internal func _NSXMLParserStartElementNs(_ ctx: _CFXMLInterface, localname: Unsa
             var namespaceNameString: String?
             var asAttrNamespaceNameString: String?
             if let ns = namespaces[idx] {
-                if reportNamespaces {
-                    namespaceNameString = UTF8STRING(ns)
-                }
-                asAttrNamespaceNameString = _colonSeparatedStringFromPrefixAndSuffix("xmlns", 5, ns, UInt(strlen(unsafeBitCast(ns, to: UnsafePointer<Int8>.self))))
+                namespaceNameString = UTF8STRING(ns)
```

Comment:
This sort of thing is what I was commenting on. Is this not making the initialization of namespaceNameString unconditional, where before it would only happen if we were reporting namespaces?